### PR TITLE
reformat-code.py: add argumentless mode for formatting diffs

### PR DESCRIPTION
**Pull request by djcampello**
This is basically implementing `git diff -U0 | clang-format-diff-11 -p1 | sed '/^+/s/g_autoptr (/g_autoptr(/g;/^+/s/sizeof (/sizeof(/g;/^+/s/g_auto (/g_auto{/g' | patch -p0`

### CODE REVIEW

**Comment by hughsie** [comment]
Could we add something in a README.md on how new contributors are supposed to use this? Thanks.

File: contrib/reformat-code.py [hunk 1]
```diff
@@ -11,31 +11,36 @@
 import subprocess
 
 CLANG_FORMATTERS = [
+    "clang-format-11",
     "clang-format-13",
     "clang-format",
 ]
+CLANG_DIFF_FORMATTERS = [
+    "clang-format-diff-11",
```
Comment (superm1) [inline contrib/reformat-code.py:19]: Do you actually want to add 11 or not? I thought there was features needed from 13.

**Comment by superm1** [comment]
> Could we add something in a README.md on how new contributors are supposed to use this? Thanks.

Since it's running on diffs it might be worth using with precommit now too.

**Comment by hughsie** [comment]
> Since it's running on diffs it might be worth using with precommit now too.

Still not a fan; can we give it a few more iterations?

File: contrib/reformat-code.py (hunk 1 above)
Comment (djcampello) [reply contrib/reformat-code.py:19]: On https://github.com/fwupd/fwupd/pull/3526 we removed the align consecutive declarations that required 13 as it not only affects structs but all declarations

**Comment by djcampello** [comment]
Added some text to CONTRIBUTING.md

**Review by hughsie — APPROVED** [review]

**Pull request merged**